services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handler.py

```diff
@@ -935,18 +947,26 @@ async def test_stop_node(
         project_id=project["uuid"], node_id=choice(all_service_uuids)  # noqa: S311
     )
     response = await client.post(f"{url}")
-    data, error = await assert_status(
+    _, error = await assert_status(
         response,
         status.HTTP_202_ACCEPTED if user_role == UserRole.GUEST else expected.accepted,
     )
-    if error is None:
-        mocked_dynamic_services_interface[
-            "dynamic_scheduler.api.stop_dynamic_service"
-        ].assert_called_once()
-    else:
-        mocked_dynamic_services_interface[
-            "dynamic_scheduler.api.stop_dynamic_service"
-        ].assert_not_called()
+
+    async for attempt in AsyncRetrying(
+        wait=wait_fixed(0.1),
+        stop=stop_after_delay(5),
+        retry=retry_if_exception_type(AssertionError),
+        reraise=True,
+    ):
+        with attempt:
```

Comment:
this is for stability checking? in the second case, if a call would happen.. then you will not check it with this call.
something like so would be needed:
```python
with contextlib.suppress(RetryError):
    async for attempt in AsyncRetrying(
        wait=wait_fixed(1),
        stop=stop_after_delay(5),
        retry=retry_unless_exception_type(AssertionError),
        reraise=True,
    ):
        with attempt:
            mocked_dynamic_services_interface[
                    "dynamic_scheduler.api.stop_dynamic_service"
                ].assert_not_called()
```

Also making retries of 0.1 for stability does not make much sense, and remember the CI is not a supra performing machine.